src/overlays/actors/ovl_En_Box/z_en_box.h

```diff
@@ -4,11 +4,84 @@
 #include <ultra64.h>
 #include <global.h>
 
+#define ENBOX_TREASURE_FLAG_UNK_MIN 20
+#define ENBOX_TREASURE_FLAG_UNK_MAX 32
+
 struct EnBox;
 
+typedef void (*EnBoxActionFunc)(struct EnBox*, GlobalContext*);
+
+typedef enum {
+    /*
+     set on init unless treasure flag is set
+     if clear, chest moves (Actor_MoveForward) (falls, likely)
+     ends up cleared from SWITCH_FLAG_FALL types when switch flag is set
+    */
+    ENBOX_STATE_IMMOBILE = 1 << 0,
+    /*
+     set in the logic for SWITCH_FLAG_FALL types
+     otherwise unused
+    */
+    ENBOX_STATE_UNUSED = 1 << 1,
+    /*
+     set with 50% chance on init for SWITCH_FLAG_FALL types
+     only used for SWITCH_FLAG_FALL types
+     ends up "blinking" (set/clear every frame) once switch flag is set,
+        if some collision-related condition (?) is met
+     only used for signum of z rotation
+    */
+    ENBOX_STATE_FALL_ANGLE_SIDE = 1 << 2,
+    /*
+     when set, is cleared next EnBox_Update call and clip to the floor
+    */
+    ENBOX_STATE_FLAG_STICK_TO_GROUND_NEXT_FRAME = 1 << 4
```

Comment:
maybe something a bit shorter like ENBOX_MOVE_GROUND or something. The specifics of it happening on the next frame can be explained in the comment